core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -195,12 +204,13 @@ public ChannelImpl shutdown() {
    * Initiates a forceful shutdown in which preexisting and new calls are cancelled. Although
    * forceful, the shutdown process is still not instantaneous; {@link #isTerminated()} will likely
    * return {@code false} immediately after this method returns.
-   *
-   * <p>NOT YET IMPLEMENTED. This method currently behaves identically to shutdown().
    */
-  // TODO(ejona86): cancel preexisting calls.
   public ChannelImpl shutdownNow() {
     shutdown();
+
+    for (ClientCallImpl<?, ?> call : activeCalls) {
+      call.getStream().cancel(Status.INTERNAL.withDescription("Shutting down"));
```

Comment:
As evidenced by the volatile stream, you need to check for `null` return value of `getStream()`. Hmm... but if it is null, it isn't guaranteed safe to ignore because the call could be between calling `clientTransportProvider.get()` (which fails after `shutdown()`) and calling `transport.newStream(..)`. Shutting down transports is non-atomic... but I think it _may_ work today. It's unclear how much we should care right now.